src/rpdk/core/extensions.py

```diff
@@ -0,0 +1,10 @@
+from .plugin_registry import get_extensions
+
+
+def setup_subparsers(subparsers, parents):
+    extensions = get_extensions()
+
+    for extension_cls in extensions.values():
+        extension = extension_cls()()
+        parser = subparsers.add_parser(extension.command_name, parents=parents)
```

Comment:
Few comments about this.

1. Is there any risk of name collision?
2. Can an extension hijack a subparser "name" and override its behavior

Reply:
Yes to both.

1. If there is a name collision, the latest one that was executed (`.add_parser`) will take precedence.
2. Given the answer from 1, yes, it is possible.

I can add error handling to verify all the options that have been added and raise an exception if there's a collision

Reply:
Added check in [37b45ce](https://github.com/aws-cloudformation/cloudformation-cli/pull/1020/commits/37b45ce2888cc2dc0ad17b919d4adbdf094037f8)